syntax = "proto2";

import "nanopb.proto";

message motd_packet {
  optional string server_name = 1 [(nanopb).max_size = 255];
  optional string motd = 2 [(nanopb).max_size = 1024];
}

message client_list_packet {
  repeated string clients = 1 [(nanopb).max_size = 16, (nanopb).max_count = 1024];
}

message d2c_envelope {
  oneof payload {
    motd_packet motd = 1;
    client_list_packet client_list = 2;
  }
}
